lib/membrane/rtp/session/receive_bin.ex

```diff
@@ -217,41 +236,31 @@ defmodule Membrane.RTP.Session.ReceiveBin do
   end
 
   @impl true
-  def handle_notification({:received_rtcp, rtcp, timestamp}, {:rtcp_parser, _ref}, state) do
+  def handle_notification({:received_rtcp, rtcp, timestamp}, {:rtcp_parser, _ref}, _ctx, state) do
     # TODO: handle RTCP reports properly
-    actions =
-      if state.receiver_reporter?,
-        do: [forward: {:receiver_reporter, {:remote_report, rtcp, timestamp}}],
-        else: []
-
-    {{:ok, actions}, state}
+    report_data = Session.Report.handle_remote_report(rtcp, timestamp, state.rtcp_report_data)
+    {:ok, %{state | rtcp_report_data: report_data}}
   end
 
   @impl true
-  def handle_notification(:send_stats, :receiver_reporter, state) do
-    remote_ssrcs = state.ssrcs |> Map.keys() |> MapSet.new()
-
-    forwards =
-      [receiver_reporter: {:ssrcs_to_report, remote_ssrcs}] ++
-        Enum.map(remote_ssrcs, &{{:rtp_stream_bin, &1}, :send_stats})
-
-    actions = Enum.map(forwards, &{:forward, &1})
-    {{:ok, actions}, state}
-  end
-
-  @impl true
-  def handle_notification({:jitter_buffer_stats, stats}, {:rtp_stream_bin, remote_ssrc}, state) do
-    actions =
-      case Map.fetch(state.ssrcs, remote_ssrc) do
-        {:ok, local_ssrc} ->
-          stats_msg = {:jitter_buffer_stats, {local_ssrc, remote_ssrc, stats}}
-          [forward: {:receiver_reporter, stats_msg}]
-
-        :error ->
-          []
-      end
-
-    {{:ok, actions}, state}
+  def handle_notification(
+        {:jitter_buffer_stats, stats},
+        {:rtp_stream_bin, remote_ssrc},
+        ctx,
+        state
+      ) do
+    {result, report_data} =
+      Session.Report.handle_stats(stats, remote_ssrc, state.ssrcs, state.rtcp_report_data)
+
+    state = %{state | rtcp_report_data: report_data}
+
+    case {result, ctx.children} do
+      {{:report, report}, %{rtcp_forwarder: _reporter}} ->
```

Comment:
This match isn't pretty. It would be much better if you could replace it with something more self descriptive like:
`if result == :ready and ctx.children |> Map.has_key(:rtcp_forwarder)`